pystac/__init__.py

```diff
@@ -4,36 +4,36 @@
 
 # flake8: noqa
 from pystac.errors import (
-    STACError,  # type:ignore
-    STACTypeError,  # type:ignore
-    ExtensionAlreadyExistsError,  # type:ignore
-    ExtensionTypeError,  # type:ignore
-    RequiredPropertyMissing,  # type:ignore
-    STACValidationError,  #  type:ignore
+    STACError,
+    STACTypeError,
+    ExtensionAlreadyExistsError,
+    ExtensionTypeError,
+    RequiredPropertyMissing,
+    STACValidationError,
```

Comment:
Noting here that the type ignores were to avoid showing errors for unused imports in VS Code with Pylance, with the `reportUnusedClass ` flag set. This was set in my configuration but not a project-level configuration, where the default Pyright behavior is to ignore this.

The better solution is to ignore `reportUnusedClass` in the VS Code Pylance settings (which is the default) and to ensure `flake8` is set as the linter for the project; that way unused imports that are not in __init__.py (which are ignored via the `# flake8: noqa` directive) will be caught in VS Code.

Reply:
Are we going to support *three* different type hint linters (mypy, pylance and pyright)? That sounds like a very difficult task, especially if one of them doesn't run in CI.

Reply:
@lossyrob My understanding of your comment is that those ignores were specific to your VS Code setup and that we are okay to drop them as @l0b0 has done here. It seems like this is mostly a PSA to other VS Code users on how to handle this. Is that right?

Might also be good to get your input on #381. My feeling is that we should settle on 1 type checker, but would be good to get other input.

Reply:
Pylance is backed by Pyright, so they are the same thing.

I had given a thumbs up on #381. I'd like to avoid a situation where Pyright (and therefore VS Code) thinks there's an error that passes CI, as it's great to be able to detect errors during editing, I find that to be the most accelerating change in my workflow in years - reminds me of when I had static compilation feedback working with other languages. However, a single type checker makes sense, and as mypy is more strict I'm hoping we A. don't get into the situation where pyright breaks, and B. some useful type-level code is blocked by mypy. Seems like that's not happening yet; also I don't think this project needs to fit to any of our personal preferences or workflows, so I think going with mypy only is a good choice.

You are correct in that my comment here was just a note in way of explination of why the `type:ignore`s were there in the first place, and a documentation of working out how to better set up VS Code for this project.